package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless

```diff
@@ -207,22 +222,33 @@ util.foreach_radio(uci, function(radio, index, config)
 	uci:set('wireless', radio_name, 'htmode', htmode)
 	uci:set('wireless', radio_name, 'country', site.regdom())
 
+	local outdoor_disabled = nil
+	if (radio.hwmode == '11a' or radio.hwmode == '11na') then
+		if is_outdoor() then
+			uci:set('wireless', radio_name, 'channels', config.outdoor_chanlist())
+			outdoor_disabled = true
+		else
+			uci:delete('wireless', radio_name, 'channels')
+		end
+	end
+
 	uci:set_list('wireless', radio_name, 'supported_rates', config.supported_rates())
 	uci:set_list('wireless', radio_name, 'basic_rate', config.basic_rate())
 
-
 	local ibss_disabled = is_disabled('ibss_' .. radio_name)
 	local mesh_disabled = is_disabled('mesh_' .. radio_name)
 
 	configure_ibss(config.ibss(), radio, index, suffix,
 		first_non_nil(
+			outdoor_disabled,
```

Comment:
Unless I'm overlooking something, this logic will never re-enable the ibss / mesh VIFs, even after outdoor mode is switched off again. I think the best solution would be to delete the ibss / mesh VIF sections completely when outdoor mode is enabled.

Reply:
Fixed in 64778022828edd97fe6da4d50690d802f84d9f7c